uswds/uswds
# Icon: Add X social media icon

# Summary

**Add X social media icon.** We added the X social media icon, but also preserved the legacy Twitter icon to allow teams to make the decision on when to update their social media information.

<img width="1154" alt="X icon as part of the icon Storybook page" src="https://github.com/uswds/uswds/assets/11464021/bcaaaeb9-9056-4852-b9f8-6ca30a8aca7c">


## Breaking change

This is not a breaking change.

## Related issue

Closes #5471

## Related pull requests

Changelog PR: https://github.com/uswds/uswds-site/pull/2329

## Preview link

[Icon Storybook page](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/dw-add-x-icon/?path=/story/design-tokens-icons--icons) → 

## Problem statement

The social media platform Twitter recently rebranded to X, and [has a new logo](https://about.twitter.com/en/who-we-are/brand-toolkit). Teams that wish to update their social media icons to reflect this change have no way to use the X icon in the same way they used the Twitter icon.

## Solution

This PR adds a X icon that conforms with the brand toolkit linked above. I've included this new icon while also maintaining the legacy Twitter icon so teams can be intentional about updating their social media account information. Just as the previous Twitter icon was named `twitter`, this new icon is named `x`. I wondered about how well a one-character name would work in our spritesheet, but it seems to work without issue. I considered renaming the icon something like `social-x` to disambiguate with something like a `close` icon, but decided that naming consistency with the other social icons was more important than solving for an issue (icon ambiguity) that we haven't actually observed.

This PR adds the new X icon to the USWDS icon set and to the default sprite.

It also adds an icon template in Sketch format so we're better able to manage icon source files going forward.

## Testing and review

On the [Icon page](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/dw-add-x-icon/?path=/story/design-tokens-icons--icons) in the USWDS Storybook:
1. Check to see that the X icon appears, in alphabetical order.
1. Check to see that the legacy Twitter icon also exists. 
2. Check that the icon changes to white on highlight as expected.

On the [Footer page](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/dw-add-x-icon/?path=/story/components-footer--default) in the USWDS Storybook:
1. Update the twitter icon image src to `src="./img/usa-icons/x.svg"`
1. Check that the X icon appears and looks about the same visual size as the legacy Twitter icon

In the `usa-icon` package:
1. Check that the `src/img/template/uswds-icon-template.sketch` file exists
